internal/report/report.go

```diff
@@ -140,8 +164,10 @@ func (writer Writer) parseReport(er engine.Report) ([]vulnerability, error) {
 
 // isExcluded returns whether the provided [report.Vulnerability] is
 // excluded based on the [Writer] configuration and the affected target.
-func (writer Writer) isExcluded(v report.Vulnerability, target string) (bool, error) {
-	for _, excl := range writer.exclusions {
+func (writer *Writer) isExcluded(v report.Vulnerability, target string) ([]int, error) {
```

Comment:
I'd rename the method so the returned values make more sense:

```suggestion
func (writer *Writer) matchExclusions(v report.Vulnerability, target string) ([]int, error) {
```